proxy/src/registry.rs

```diff
@@ -204,6 +203,12 @@ pub struct User {
 /// Methods to interact with the Registry in a uniform way.
 #[async_trait]
 pub trait Client: Clone + Send + Sync {
+    /// Fetch latest height by virtue of checking the block header of the best chain.
+    ///
+    /// # Errors
+    ///
+    /// Will return `Err` if a protocol error occurs.
+    async fn latest_height(&self) -> Result<u32, error::Error>;
```

Comment:
This should maybe be called `chain_height` or `best_height`, I think using "latest" could be confusing, because it's not a matter of time.